bin/get_metadata.py

```diff
@@ -109,18 +118,25 @@ def get_read_cycles(self):
         read_counter = 1
         index_counter = 1
         try:
-            for read_info in self.stats_json["ReadInfosForLanes"][0]["ReadInfos"]:
-                if read_info["IsIndexedRead"]:
+            reads = self.run_info["RunInfo"]["Run"]["Reads"]["Read"]
+
+            # Handle potential cases with only one Read element
+            if isinstance(reads, dict):
+                reads = [reads]  # Wrap it in a list to make it iterable
+
+            for read_info in reads:
```

Comment:
I'm not sure if we can assume that the list of read entries always is sorted. Perhaps you could sort them by "@Number"?